agent/agent.go

```diff
@@ -304,13 +304,15 @@ type processNextResponse struct {
 }
 
 func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskResponse, tc *taskContext, needTeardownGroup bool) (processNextResponse, error) {
+	processNextTaskCtx, span := a.tracer.Start(ctx, "process-next-task")
```

Comment:
if we aren't planning to propagate the span down the chain with the ctx we could potentially ignore the returned ctx, like
```go
_, span := a.tracer.Start(ctx, "process-next-task")
```

Reply:
done